src/org/jetbrains/plugins/scala/lang/refactoring/util/ScalaRefactoringUtil.scala

```diff
@@ -139,6 +139,33 @@ object ScalaRefactoringUtil {
     Option(element).filter(_.getTextRange.getEndOffset == endOffset).flatMap(checkTypeElement)
   }
 
+  def isTypeContainsTypeParameter(typeElement: ScTypeElement): Boolean = {
+    typeElement.breadthFirst.forall {
+      case x: ScTypeElement if x.calcType.isInstanceOf[ScTypeParameterType] =>
+        return true
+      case _ => true
+    }
+    false
+  }
+
```

Comment:
Probably it would be better to find reference in the type element and resolve it.